### ISSUE TITLE
[BERT] Functionality for BERT Pre-training 

### ISSUE THREAD

**Issue by eric-haibin-lin** [dmlc/gluon-nlp#475]
- [x] support `truncated_normal` for better initialization for pre-training https://www.tensorflow.org/api_docs/python/tf/initializers/truncated_normal. We can follow the pytorch implementation https://github.com/pytorch/pytorch/pull/5015/files . And add inverse err function operator in MXNet:  https://github.com/apache/incubator-mxnet/pull/13229 
- [x] support Adam with fixed weight decay 
- [ ] Update compare_tf_gluon.py with pre-training output for masked language model and next sentence prediction
- [ ] Training and evaluation script

**Comment by eric-haibin-lin** [comment]
Would someone be interested to take `truncated_normal`? I can PR the implementation for inverse err operator to MXNet, and the rest can be done in python I think

**Comment by eric-haibin-lin** [comment]
#482 for adam with weight decay

**Comment by Ishitori** [comment]
I will work on `truncated_normal`.

**Comment by Ishitori** [comment]
Submitted PR for truncnorm - https://github.com/dmlc/gluon-nlp/pull/548

**Comment by sravanbabuiitm** [comment]
From Comprehend perspective, I have observed Text classification performance to be not so great for extremely short documents. This might be due to the fact that BERT is pretrained on long docs of wikipedia. It would definitely help comprehend custom classification in trying out pretraining BERT on additional data sources which cater additional domains like Twitter, Amazon reviews, kindle books & other noisy settings.
